src/pages/add-lock-exception/AddLockExceptionForm/OrganisationUnitSelection/OrganisationUnitSelection.js

```diff
@@ -0,0 +1,154 @@
+        lazy: true,
+    })
+
+    useEffect(() => {
+        setCurrentRoot(null)
+        refetch({ dataSetId })
+    }, [dataSetId])
+
+    const handleOrgUnitClick = ({ selected, ...currentRoot }) => {
+        setCurrentRoot(currentRoot)
+        onSelectedChange(selected)
+    }
+    const handleSelect = orgUnitPaths => {
+        const newSelected = union(selected, orgUnitPaths)
+        onSelectedChange(newSelected)
+    }
+    const handleDeselect = orgUnitPaths => {
+        const newSelected = difference(selected, orgUnitPaths)
+        onSelectedChange(newSelected)
+    }
+
+    if (loading || !called) {
+        return (
+            <CenteredContent>
+                <CircularLoader />
+            </CenteredContent>
+        )
+    }
+
+    if (error) {
+        return (
+            <NoticeBox title={i18n.t('Error loading data for data set')} error>
+                {error.message}
+            </NoticeBox>
+        )
+    }
+
+    const rootWithMembers = data.rootWithMembers.organisationUnits[0]
+    const orgUnitPaths = data.dataSetMembers.organisationUnits.map(
+        ou => ou.path
+    )
+
+    return (
+        <div className={styles.container}>
+            <div
+                className={styles.organisationUnitTree}
+                data-test="add-lock-exception-org-unit-tree"
+            >
+                <OrganisationUnitTree
+                    roots={[rootWithMembers.id]}
+                    initiallyExpanded={[rootWithMembers.path]}
+                    highlighted={currentRoot ? [currentRoot.id] : []}
+                    filter={orgUnitPaths}
+                    selected={selected}
+                    onChange={handleOrgUnitClick}
+                />
+            </div>
+            <div className={styles.right}>
+                <div className={styles.currentRoot}>
+                    {currentRoot
+                        ? i18n.t(
+                              'For organisation units within {{organisationUnitName}}:',
+                              {
+                                  organisationUnitName: currentRoot.displayName,
+                              }
+                          )
+                        : i18n.t('For all organisation units:')}
+                </div>
+                <SelectByLevel
+                    levels={levels}
+                    currentRoot={currentRoot}
+                    onSelect={handleSelect}
+                    onDeselect={handleDeselect}
+                />
+                <SelectByGroup
+                    groups={groups}
+                    currentRootId={currentRoot?.id}
+                    onSelect={handleSelect}
+                    onDeselect={handleDeselect}
+                />
+                <SelectAll
+                    currentRootId={currentRoot?.id}
+                    allOrgUnitPaths={orgUnitPaths}
+                    onSelect={handleSelect}
+                    onDeselect={handleDeselect}
+                />
```

Comment:
These components are rewrites of the [`OrgUnitSelect` components from `d2-ui`](https://github.com/dhis2/d2-ui/tree/d0d3812fe70b8b6f6c33cf0f85d89ba987f8b8c1/packages/org-unit-select/src) that use `useDataQuery` instead of `d2`.